core/src/main/scala/newtypes.scala

```diff
@@ -23,36 +41,36 @@ object newtypes:
-    inline def apply(inline s: Impl): Newtype            = s.asInstanceOf[Newtype]
-    inline def from[M[_]](inline f: M[Impl]): M[Newtype] = f.asInstanceOf[M[Newtype]]
-    inline def from[M[_], B](using sr: SameRuntime[B, Impl])(inline f: M[B]): M[Newtype] =
+  abstract class TotalWrapper[Newtype, Impl](using Newtype =:= Impl):
+    inline final def raw(inline a: Newtype): Impl              = a.asInstanceOf[Impl]
+    inline final def apply(inline s: Impl): Newtype            = s.asInstanceOf[Newtype]
+    inline final def from[M[_]](inline f: M[Impl]): M[Newtype] = f.asInstanceOf[M[Newtype]]
+    inline final def from[M[_], B](using sr: SameRuntime[B, Impl])(inline f: M[B]): M[Newtype] =
       f.asInstanceOf[M[Newtype]]
-    inline def from[M[_], B](inline other: TotalWrapper[B, Impl])(inline f: M[B]): M[Newtype] =
+    inline final def from[M[_], B](inline other: TotalWrapper[B, Impl])(inline f: M[B]): M[Newtype] =
       f.asInstanceOf[M[Newtype]]
-    inline def raw[M[_]](inline f: M[Newtype]): M[Impl] = f.asInstanceOf[M[Impl]]
+    inline final def raw[M[_]](inline f: M[Newtype]): M[Impl] = f.asInstanceOf[M[Impl]]
 
-    given SameRuntime[Newtype, Impl]    = identity
-    given SameRuntime[Impl, Newtype]    = _.asInstanceOf[Newtype]
-    given (using Eq[Impl]): Eq[Newtype] = Eq.by(_.value)
+    given SameRuntime[Newtype, Impl] = raw(_)
+    given SameRuntime[Impl, Newtype] = apply(_)
+    given (using e: Eq[Impl]): Eq[Newtype] = new Eq[Newtype]:
+      override def eqv(x: Newtype, y: Newtype) = e.eqv(raw(x), raw(y))
 
-    extension (a: Newtype)
-      inline def value: Impl                                     = a
-      inline def into[X](inline other: TotalWrapper[X, Impl]): X = other.apply(a)
-      inline def map(inline f: Impl => Impl): Newtype            = apply(f(a))
+    extension (inline a: Newtype)
+      inline def value: Impl                                     = raw(a)
+      inline def into[X](inline other: TotalWrapper[X, Impl]): X = other.apply(raw(a))
+      inline def map(inline f: Impl => Impl): Newtype            = apply(f(raw(a)))
   end TotalWrapper
 
-  trait FunctionWrapper[Newtype, Impl](using ev: Newtype =:= Impl) extends TotalWrapper[Newtype, Impl]:
-    extension (a: Newtype) inline def apply: Impl = a
+  abstract class FunctionWrapper[Newtype, Impl](using Newtype =:= Impl) extends TotalWrapper[Newtype, Impl]:
+    extension (inline a: Newtype) inline def apply: Impl = a.asInstanceOf[Impl]
 
-  trait OpaqueString[A](using A =:= String) extends TotalWrapper[A, String]:
+  abstract class OpaqueString[A](using A =:= String) extends TotalWrapper[A, String]:
     given Show[A]   = _.value
     given Render[A] = _.value
 
-  trait OpaqueInt[A](using A =:= Int) extends TotalWrapper[A, Int]:
+  abstract class OpaqueInt[A](using A =:= Int) extends TotalWrapper[A, Int]:
     extension (inline a: A)
-      inline def unary_- : A                      = apply(-raw(a))
```

Comment:
We also need to restore this

Reply:
Yeah that's a mistake. I was trying to extend OpaqueInt from OpaqueIntSafer but it caused errors, so I reverted but missed this